### DISCUSSION TITLE
CORS가 무엇인지 설명해주세요.

### DISCUSSION THREAD

**Discussion by yeonkkk** [woowacourse-study/2023-tal-study#64] [기술 면접 질답 모음집]
CORS 에 대해 설명해주세요.

**Comment by yeonkkk** [comment]
CORS는 출처가 다른 자원들을 공유한다는 뜻으로, 한 출처에 있는 자원에서 다른 출처에 있는 자원에 접근하도록 하는 개념입니다. 

다른 출처로부터 조회된 자원들의 읽기 접근을 막아 다른 출처 공격을 예방하기 위한 동일 출처 정책으로 다른 출처 리소스에 접근하기에 어려움이 있습니다. 이에 다른 출처 리소스에 접근성을 높이기 위한 방법으로 CORS가 등장하게 되었습니다.

주요 정책으로는 단순요청, 프리플라이트 요청, 인증정보요청 으로 3가지가 있습니다.

**Reply by yoondgu** [reply]
CORS는 웹 애플리케이션이 다른 출처에서 자원에 대한 요청을 했을 때, 이에 대해 접근할 수 있는 권한을 부여하도록 HTTP 헤더를 사용해 브라우저에 알려주는 메커니즘이다.
CORS 헤더를 활용해 SOP 위반 여부 및 허용 여부를 확인할 수 있다.

SOP란 Same Origin Policy로, 다른 출처의 리소스를 사용하는 것을 제안하는 보안 정책이다. 다른 출처의 사용자가 스크립트를 이용해 잘못된 요청을 보낼 수 있기 때문이다.
이 때 url의 프로토콜, 호스트, 포트가 같아야 같은 출처라고 판단한다.
(localhost와 127.0.0.1은 같은 출처인가? no)

CORS 요청에는 단순 요청, 프리플라이트 요청, 인증정보 포함(Credentialed) 요청 세 가지가 있다.
